src/pages/workspace/tags/WorkspaceViewTagsPage.tsx

```diff
@@ -0,0 +1,291 @@
+    }, [isFocused]);
+
+    const policyTagLists = useMemo(() => PolicyUtils.getTagLists(policyTags).filter((policy) => policy.name === currentTagListName), [currentTagListName, policyTags]);
+    const tagList = useMemo<PolicyForList[]>(
+        () =>
+            policyTagLists
+                .map((policyTagList) =>
+                    lodashSortBy(Object.values(policyTagList.tags || []), 'name', localeCompare).map((value) => {
+                        const tag = value as OnyxCommon.OnyxValueWithOfflineFeedback<OnyxTypes.PolicyTag>;
+                        const isDisabled = tag.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
+                        return {
+                            value: tag.name,
+                            text: PolicyUtils.getCleanedTagName(tag.name),
+                            keyForList: tag.name,
+                            isSelected: !!selectedTags[tag.name],
+                            pendingAction: tag.pendingAction,
+                            errors: tag.errors ?? undefined,
+                            enabled: tag.enabled,
+                            isDisabled,
+                            rightElement: <RightElementEnabledStatus enabled={tag.enabled} />,
+                        };
+                    }),
+                )
+                .flat(),
+        [policyTagLists, selectedTags],
+    );
+
+    if (!currentPolicyTag) {
+        return <NotFoundPage />;
+    }
+
+    const tagListKeyedByName = tagList.reduce<Record<string, PolicyForList>>((acc, tag) => {
+        acc[tag.value] = tag;
+        return acc;
+    }, {});
+
+    const toggleTag = (tag: PolicyForList) => {
+        setSelectedTags((prev) => ({
+            ...prev,
+            [tag.value]: !prev[tag.value],
+        }));
+    };
+
+    const toggleAllTags = () => {
+        const isAllSelected = tagList.every((tag) => !!selectedTags[tag.value]);
+        setSelectedTags(isAllSelected ? {} : Object.fromEntries(tagList.map((item) => [item.value, true])));
+    };
+
+    const getCustomListHeader = () => (
+        <View style={[styles.flex1, styles.flexRow, styles.justifyContentBetween, styles.pl3, styles.pr9]}>
+            <Text style={styles.searchInputStyle}>{translate('common.name')}</Text>
+            <Text style={[styles.searchInputStyle, styles.textAlignCenter]}>{translate('statusPage.status')}</Text>
+        </View>
+    );
+
+    const navigateToTagSettings = (tag: PolicyOption) => {
+        Navigation.navigate(ROUTES.WORKSPACE_TAG_SETTINGS.getRoute(policyID, tag.keyForList));
+    };
+
+    const selectedTagsArray = Object.keys(selectedTags).filter((key) => selectedTags[key]);
+
+    const handleDeleteTags = () => {
+        setSelectedTags({});
+        Policy.deletePolicyTags(policyID, selectedTagsArray);
+        setDeleteTagsConfirmModalVisible(false);
+    };
+
+    const isLoading = !isOffline && policyTags === undefined;
+
+    const getHeaderButtons = () => {
```

Comment:
This function looks like it has a lot of redundant looping. We can compute it all in one pass, and I think the result is simpler and more readable too:

```ts
const getHeaderButtons = () => {
    const options: Array<DropdownOption<DeepValueOf<typeof CONST.POLICY.TAGS_BULK_ACTION_TYPES>>> = [];

    if (selectedTagsArray.length === 0) {
        return null;
    }

    options.push({
        icon: Expensicons.Trashcan,
        text: translate(selectedTagsArray.length === 1 ? 'workspace.tags.deleteTag' : 'workspace.tags.deleteTags'),
        value: CONST.POLICY.TAGS_BULK_ACTION_TYPES.DELETE,
        onSelected: () => setDeleteTagsConfirmModalVisible(true),
    });

    let enabledTagCount = 0;
    const tagsToDisable: Record<string, {name: string; enabled: boolean}> = {};
    let disabledTagCount = 0;
    const tagsToEnable: Record<string, {name: string; enabled: boolean}> = {};
    for (const tagName of selectedTagsArray) {
        if (tagListKeyedByName[tagName]?.enabled) {
            enabledTagCount++;
            tagsToDisable[tagName] = {
                name: tagName,
                enabled: false,
            };
        } else {
            disabledTagCount++;
            tagsToEnable[tagName] = {
                name: tagName,
                enabled: true,
            };
        }
    }

    if (enabledTagCount > 0) {
        options.push({
            icon: Expensicons.DocumentSlash,
            text: translate(enabledTagCount === 1 ? 'workspace.tags.disableTag' : 'workspace.tags.disableTags'),
            value: CONST.POLICY.TAGS_BULK_ACTION_TYPES.DISABLE,
            onSelected: () => {
                setSelectedTags({});
                Policy.setWorkspaceTagEnabled(policyID, tagsToDisable);
            },
        });
    }

    if (disabledTagCount > 0) {
        options.push({
            icon: Expensicons.Document,
            text: translate(disabledTagCount === 1 ? 'workspace.tags.enableTag' : 'workspace.tags.enableTags'),
            value: CONST.POLICY.TAGS_BULK_ACTION_TYPES.ENABLE,
            onSelected: () => {
                setSelectedTags({});
                Policy.setWorkspaceTagEnabled(policyID, tagsToEnable);
            },
        });
    }

    return (
        <ButtonWithDropdownMenu
            buttonRef={dropdownButtonRef}
            onPress={() => null}
            shouldAlwaysShowDropdownMenu
            pressOnEnter
            isSplitButton={false}
            buttonSize={CONST.DROPDOWN_BUTTON_SIZE.MEDIUM}
            customText={translate('workspace.common.selected', {selectedNumber: selectedTagsArray.length})}
            options={options}
            style={[isSmallScreenWidth && styles.flexGrow1, isSmallScreenWidth && styles.mb3]}
        />
    );
};
```